### DISCUSSION TITLE
trait / unobserved heterogeneity effect for type=’stanct'

### DISCUSSION THREAD

**Discussion by gab-pr** [cdriveraus/ctsem#60] [Q&A] [answered]
Hi Charles,
I would like to run a bivariate ctm estimating within and between effects by modeling unobserved heterogeneity (TRAITVAR = "auto"). However, I see that TRAITVAR is for type=’omx’ only. How can I model unobserved heterogeneity when using type = "stanct" and ctStanFit?
Thank you very much
Gabriele

**Comment by cdriveraus** [comment]
Hi Gabriele, 
From the manual :) "By default, intercept related model parameters (T0MEANS, MANIFESTMEANS, CINT) are set
to individually varying using a correlated random effects formulation, while other parameters are
fixed across subjects."
With the change away from OpenMx, individual variation on all parameters became possible, so TRAITVAR was removed in favour of a more general approach. Page 5 of the manual `ctDocs()` or https://cran.r-project.org/web/packages/ctsem/vignettes/hierarchicalmanual.pdf discusses model specification.

**Reply by cdriveraus** [reply]
In case it's not clear, TRAITVAR was simply random effects on the continuous intercept, which is now the default but can be disabled as per the model spec section.

**Comment by gab-pr** [comment]
Hi Charles,
thank you very much for your prompt reply. If I understand correctly, when using type = "stanct" and ctStanFit, unobserved heterogeneity or between subject differences are controlled for by default. 
Thank you again
Gabriele

**Reply by cdriveraus** [reply, marked as answer]
correct! By default the MANIFESTMEANS (measurement intercept) are free, so in case you are using multiple indicator factor models you may want to consider if individual variation on each indicator is what you desire, but with one indicator they are equivalent whether it's on the latent intercept (CINT) or measurement.
